internal/qa/testing.go

```diff
@@ -280,8 +280,8 @@ func fixHCL(v interface{}) interface{} {
 	}
 }
 
-// EnvironmentTemplate asserts existence and fills in {env.VAR} & {var.RANDOM} placeholders in template
-func EnvironmentTemplate(t *testing.T, template string) string {
+// For writing a unit test to intercept the errors (t.Fatalf literally ends the test in failure)
+func environmentTemplate(t *testing.T, template string) (string, error) {
```

Comment:
where this private method is going to be used?

Reply:
it was refactored to write the unit test otherwise t.Fatal makes it untestable to test the unpopulated logic. It is only used in the EnvironmentTemplate so it is kept as a private method to the package. No one else should be calling it. It is only used to unit test the error logic.